crates/brainfuck_prover/src/components/io/table.rs

```diff
@@ -77,6 +85,55 @@ impl<const N: u32> IOTable<N> {
             self.add_row(dummy_row);
         }
     }
+
+    /// Transforms the [`IOTable`] into a [`TraceEval`], to be committed when
+    /// generating a STARK proof.
+    ///
+    /// The [`IOTable`] is transformed from an array of rows (one element = one step
+    /// of all registers) to an array of columns (one element = all steps of one register).
+    /// It is then evaluated on the circle domain.
+    ///
+    /// # Returns
+    /// A tuple containing the evaluated trace and claim for STARK proof.
+    ///
+    /// # Errors
+    /// Returns [`TraceError::EmptyTrace`] if the table is empty.
+    pub fn trace_evaluation(&self) -> Result<(TraceEval, Claim), TraceError> {
+        let n_rows = self.table.len() as u32;
+
+        // It is possible that the table is empty because the program has no input or output.
+        //
+        // In this case, we return:
+        // - An empty trace.
+        // - A claim with a log size of 0.
```

Comment:
This part can be refactored into the `# Returns` part of the header doc

Reply:
fixed